docs/brillig/00_intro.md

```diff
@@ -0,0 +1,47 @@
+# Intro to Brillig
+BRILLIG - Bytecode Runtime Interface for Liberated Language Implementation... G
+
+The G stands for whatever you like, like [Gabagool](https://en.wikipedia.org/wiki/Gabagool).
+
+Noir can compile to bytecode for Brillig, a general VM architecture for use with Noir.
+It is used for functions that are unconstrained, non-deterministic, or otherwise need a VM. 
+
+Why we need Brillig
+---
+
+Noir is a zero-knowledge language that specializes in zero-knowledge proofs that flexibly can compile code to constraints. Despite its specialization, Noir still requires a general VM for the following reasons:
```

Comment:
Again I think we should reword this from saying what Noir needs, but rather, what _any_ NP complete circuit language (such as ACIR) would require in order to enable the features listed (unconstrained execution, etc)